pkg/watch/health.go

```diff
@@ -0,0 +1,257 @@
+package watch
+
+import (
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/square/p2/pkg/health"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/preparer"
+	"github.com/square/p2/pkg/util/net"
+)
+
+// These constants should probably all be something the p2 user can set
+// in their preparer config...
+
+// number of milliseconds between reality store checks
+const POLL_KV_FOR_PODS = 2000
+
+// number of milliseconds between health checks
+const HEALTHCHECK_INTERVAL = 1000
+
+// TTL on healthchecks in seconds
+const TTL = 60
+
+// Contains method for watching the consul reality store to
+// track services running on a node. A manager method:
+// WatchHealth tracks the reality store and manages
+// a health checking go routine for each service in the
+// reality store
+
+// PodWatch houses a pod's manifest, a channel to kill the
+// pod's goroutine if the pod is removed from the reality
+// tree, and a bool that indicates whether or not the pod
+// has a running MonitorHealth go routine
+type PodWatch struct {
+	manifest kp.ManifestResult
```

Comment:
I noticed you use `manifest.Manifest` a lot of places in this code. Probably a sign that this should just be a `*pods.Manifest` instead of a `kp.ManifestResult`